portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,36 +37,56 @@
 /** Servlet that returns data for my portfolio*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  private List<String> output = new ArrayList<>();
-  
+  private static final int COMMENT_LIMIT_DEFAULT = 5;
+  private static final String COMMENT_LANGUAGE_CODE_DEFAULT = "en";
+  private Comments comments = new Comments(COMMENT_LIMIT_DEFAULT,
+                                           new ArrayList<String>(),
+                                           COMMENT_LANGUAGE_CODE_DEFAULT);
```

Comment:
I don't think you need to cache the comments in memory: for a Get request, read from the datastore and create local comments and return; for a Post request, you can either cache the limit/language as local variables for persistently to datastore.